straxen/plugins/event_processing.py

```diff
@@ -733,58 +733,175 @@ def cs2_to_e(self, x):
 
 
 @export
+@strax.takes_config(
+    strax.Option(name='pre_s1_area_threshold', default=1e3,
+                 help='Only take S1s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='pre_s2_area_threshold', default=1e4,
+                 help='Only take S2s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='deltatime_exponent', default=-1.0,
+                 help='The exponent of delta t when calculating shadow'),
+    strax.Option(name='time_window_backward', default=int(1e9),
+                 help='Search for peaks casting shadow in this time window [ns]')
+)
 class EventShadow(strax.Plugin):
     """
     This plugin can calculate shadow at event level.
     It depends on peak-level shadow.
-    The event-level shadow is its first S2 peak's shadow.
-    If no S2 peaks, the event shadow will be nan.
-    It also gives the position infomation of the previous S2s
+    The event-level shadow is its first peak peak's shadow.
```

Comment:
sorry, it's not clear to me the definition of "event-level shadow". Can you please elaborate a bit on this explanation?

Reply:
I just updated the docstring. An event basically has three shadows cast by the primary S2 peak. They are: 

1. Main S1's shadow
2. Main S2's shadow
3. The first S2's shadow

The third one is a little bit confusing. But given the max drift time is ~2ms and we select a 1 sec time window when calculating shadow, there is a probability that the main S2's shadow is from the first S2 if the first S2 is larger than 1e4 PE(the threshold for 'large' peak). Also considering some events have no S1, an event's S2 shadow is its first S2's shadow.